lib/get-transitive-data.js

```diff
@@ -368,66 +395,52 @@ export function bundle ({data, log}) {
     }
 
     log(`dissimilarity ${minDis}`)
-    if (minDis > MAXIMUM_DISSIMILARITY) break
+    if (minDis > MAX_DISSIMILARITY) break
 
     // cluster the least dissimilar clusters
     clusters[minI] = clusters[minI].concat(clusters[minJ])
     clusters.splice(minJ, 1) // remove clusters[j]
   }
 
+  log(`replaced ${inputSize} journeys with ${clusters.length} clusters`)
   // merge journeys together
-  data.journeys = clusters.map(c => {
+  return clusters.map((c) => {
     return c.reduce((j1, j2) => {
       for (let i = 0; i < j1.segments.length; i++) {
         if (j1.segments[i].type !== 'TRANSIT') continue
 
         // convert to pattern groups
         if (!j1.segments[i].patterns) {
-          j1.segments[i].patterns = [{
-            pattern_id: j1.segments[i].pattern_id,
-            from_stop_index: j1.segments[i].from_stop_index,
-            to_stop_index: j1.segments[i].to_stop_index
-          }]
-          j1.segments[i].pattern_id = j1.segments[i].from_stop_index = j1.segments[i].to_stop_index = undefined
+          j1.segments[i].patterns = [{...j1.segments[i]}]
+          j1.segments[i].pattern_id =
+            j1.segments[i].from_stop_index =
+            j1.segments[i].to_stop_index =
+            undefined
         }
 
         // don't modify from and to indices, Transitive will use the stops from the first pattern
         // TODO replace with "places" (e.g. "Farragut Square Area")
-        j1.segments[i].patterns.push({
-          pattern_id: j2.segments[i].pattern_id,
-          from_stop_index: j2.segments[i].from_stop_index,
-          to_stop_index: j2.segments[i].to_stop_index
-        })
+        j1.segments[i].patterns.push({...j2.segments[i]})
       }
 
       return j1
     })
   })
-
-  // for now just return the first journey
-  // data.journeys = clusters.map(c => c[0])
-
-  // color routes according to cluster
-  const color = hsl('#b00')
-  clusters.forEach((c) => {
-    color.h += 67
-    const rgb = color + ''
-    c.forEach((j) => {
-      j.segments.forEach((s) => {
-        s.color = rgb
-      })
-    })
-  })
-
-  log(`replaced ${inputSize} journeys with ${clusters.length} clusters`)
 }
 
 /**
- * return the dissimilarity between two clusters, using complete linkages
- * (see James et al., _An Introduction to Statistical Learning, with Applications in R_. New York: Springer, 2013, pg. 395.)
+ * Get the dissimilarity between two clusters, using complete linkages (see
+ * James et al., _An Introduction to Statistical Learning, with Applications in
+ * R_. New York: Springer, 2013, pg. 395.)
  */
-function clusterDissimilarity (c1, c2, data) {
-  let ret = 0
+function getClusterDissimilarity (c1: Array<Journey>, c2: Array<Journey>, {
+  patterns,
+  stops
+}: {
+  patterns: Array<Pattern>,
+  stops: Array<Stop>
+}): number {
+  let dissimilarity = 0
 
   for (const j1 of c1) {
```

Comment:
Ugh. Who wrote code with variables like this 🙈  (it was me). I'll refactor next week.